src/bin/common/solving_loop.ml

```diff
@@ -955,10 +985,18 @@ let main () =
             st
         end
 
-      | {contents = `Other (custom, args); _} ->
-        handle_custom_statement custom args st
+      | {contents = `Other (custom, args); loc; _} ->
+        handle_custom_statement loc custom args st
 
-      | _ ->
+      | td ->
+        let st =
+          match td.contents with
+          | `Pop _ ->
+            State.set has_incremental (State.get has_incremental st - 1) st
+          | `Push _ ->
+            State.set has_incremental (State.get has_incremental st + 1) st
```

Comment:
This is not +1 and -1; Push and Pop have an integer argument which indicates how many levels are pushed/popped.

```suggestion
          | `Pop n ->
            State.set has_incremental (State.get has_incremental st - n) st
          | `Push n ->
            State.set has_incremental (State.get has_incremental st + n) st
```